twilio/rest/__init__.py

```diff
@@ -116,11 +119,23 @@ def request(self, method, uri, params=None, data=None, headers=None, auth=None,
         if 'Accept' not in headers:
             headers['Accept'] = 'application/json'
 
-        if self.region:
-            head, tail = uri.split('.', 1)
+        if self.region or self.edge:
+            pieces = uri.split('.')
+            pieces[-2:] = ['.'.join(pieces[-2:])]
+
+            region = None
+            edge = None
+            if len(pieces) == 3:
+                region = pieces[1]
+            elif len(pieces) == 4:
+                edge = pieces[1]
+                region = pieces[2]
+
+            edge = self.edge or edge
+            region = self.region or region or (edge and 'us1')
+            pieces[1:-1] = [edge, region]
 
-            if not tail.startswith(self.region):
-                uri = '.'.join([head, self.region, tail])
+            uri = '.'.join([piece for piece in pieces if piece])
```

Comment:
Prefer to use the named parts rather than update `pieces` multiple times so this can become `uri = '.'.join([prefix, edge, region, suffix])` or whatever with the filtering.